Model.cpp

```diff
@@ -92,41 +100,46 @@ int Model::insertBatch(vector<Model*> models, int batchsize, Mode mode)
             if (!ignoreField) {
                 paramNum++;
                 
-                if (f->getType() == INTEGER)
-                {
-                    IntegerField * i = static_cast <IntegerField*>(f);
-                    if (i->isNull()) {
-                        conn.setNull(paramNum);
-                    } else {
-                        conn.setInt(paramNum, i->getValue());
-                    }
-                }
-                else if(f->getType() == FLOAT)
-                {
+                if (f->isNull()) {
+                    conn.setNull(paramNum);
+                } else {
                     
-                    FloatField * i = static_cast <FloatField*>(f);
-                    if (i->isNull()) {
-                        conn.setNull(paramNum);
-                    } else {
-                        conn.setDouble(paramNum, i->getValue());
-                    }
-                }
-                else if(f->getType() == TEXT)
-                {
-                    TextField * i = static_cast <TextField*>(f);
-                    if (i->isNull()) {
-                        conn.setNull(paramNum);
-                    } else {
-                        conn.setString(paramNum, i->getValue());
-                    }
-                }
-                else if(f->getType() == CHAR)
-                {
-                    CharField * i = static_cast <CharField*>(f);
-                    if (i->isNull()) {
-                        conn.setNull(paramNum);
-                    } else {
-                        conn.setString(paramNum, i->getValue());
+                    switch(f->getType()){
+
+                        case INTEGER: 
+                        {
```

Comment:
If I remove the brackets (and nothing else), it throws errors along the lines of `error: jump to case label [-fpermissive]`,` error:   crosses initialization of ‘IntegerField* i’`  and ` error: conflicting declaration ‘BooleanField* i’ `.

This seems to have something to do with declaring variables within a switch statement, which is not allowed for reasons I don't entirely follow (see below...), but they recommend adding an additional layer of scope to fix this problem. 

https://stackoverflow.com/questions/449736/why-can-t-variables-be-declared-in-a-switch-statement